### PR TITLE
session: print txn_mode in the general log

### PR DESCRIPTION
**Pull request by sticnarf**
### What problem does this PR solve? 

Currently, we cannot know in which mode (optimistic or pessimistic) a transaction is running. It could make some trouble when diagnosing problems.

### What is changed and how it works?

Print `txn_mode` in the general log.

### Check List 

Tests 

- No need IMO.

Code changes

 - Print more logs

Side effects

- Increase log size when general log in on.

Related changes

 - Probably cherry pick to 3.0 and 3.1?

Release note

 - Unnecessary.

### CODE REVIEW

**Comment by sticnarf** [comment]
** Note: I am not sure whether this PR will affect other tools analysing logs **

**Comment by zz-jason** [comment]
@sticnarf please add some proper labels for this PR.

File: server/conn.go [hunk 1]
```diff
@@ -690,6 +690,7 @@ func (cc *clientConn) Run(ctx context.Context) {
 				zap.String("command", mysql.Command2Str[data[0]]),
 				zap.String("status", cc.SessionStatusToString()),
 				zap.String("sql", queryStrForLog(string(data[1:]))),
+				zap.String("txn_mode", cc.ctx.GetSessionVars().GetReadableTxnMode()),
```
Comment (sticnarf) [inline server/conn.go]: Help me check if there is risk of nil pointer dereference

Comment (zz-jason) [reply server/conn.go]: any update?

Comment (jackysp) [reply server/conn.go]: I think it could not be nil, PTAL @coocood .

Comment (coocood) [reply server/conn.go]: if handleChangeUser failed, cc.ctx may be nil

**Comment by coocood** [comment]
LGTM

**Pull request merged**